subsys: bluetooth: services: add continuous glucose monitoring service (#7910)
nrfconnect/sdk-nrf: doc/nrf/libraries/bluetooth_services/services/cgms.rst

.. _cgms_readme:

GATT Continuous Glucose Monitoring Service (CGMS)
#################################################

.. contents::
   :local:
   :depth: 2

> This module implements the Continuous Glucose Monitoring Service with the corresponding set of characteristics defined in the `Continuous Glucose Monitoring Service Specification`_.

Reviewer: Well, we use this nomenclature in other service RST files (see BMS, HIDS). I guess we should stick to the module term so that it is aligned with other service docs

.. _cgms_readme:

GATT Continuous Glucose Monitoring Service (CGMS)
#################################################

.. contents::
   :local:
   :depth: 2

Overview
********

This module implements the Continuous Glucose Monitoring Service with the corresponding set of characteristics defined in the `Continuous Glucose Monitoring Service Specification`_.

Supported features
==================

All the mandatory CGMS features are supported.

Configuration
*************

Set the maximum number of glucose measurement records stored in the device using the :kconfig:option:`CONFIG_BT_CGMS_MAX_MEAS_RECORD` Kconfig option.
The value of should be large enough to hold all records generated in a session.

Set the logging level of the CGMS module using the :kconfig:option:`CONFIG_BT_CGMS_LOG_LEVEL` Kconfig option.

Usage
*****

To use CGMS in your application, call the :c:func:`bt_cgms_init` function.

> Then, call the :c:func:`bt_cgms_add_measurement` function to pass the measurement result of glucose concentration to CGMS.

Author: It seems simply a rendering style, not a reference.

> Then, call the :c:func:`bt_cgms_add_measurement` function to pass the measurement result of glucose concentration to CGMS.

Reviewer 2: They will be references as well, as soon as the new functions are in.

.. _cgms_readme:

GATT Continuous Glucose Monitoring Service (CGMS)
#################################################

.. contents::
   :local:
   :depth: 2

Overview
********

This module implements the Continuous Glucose Monitoring Service with the corresponding set of characteristics defined in the `Continuous Glucose Monitoring Service Specification`_.

Supported features
==================

All the mandatory CGMS features are supported.

Configuration
*************

:kconfig:option:`BT_CGMS_NUMBER_MEAS_RECORD_MAX`
 Set the maximum number of glucose measurement records stored in the device.
 The value of should be large enough to hold all records generated in a session.

:kconfig:option:`CONFIG_BT_CGMS_LOG_LEVEL`
 Set the logging level of CGMS module.

Usage
*****

To use CGMS in you application, :c:func:`bt_cgms_init` should be called.

> Then, the measurement result of glucose concentration should be passed to
> CGMS by calling :c:func:`bt_cgms_add_measurement`.

Reviewer 2: ```suggestion
Then, call the :c:func:`bt_cgms_add_measurement` function to pass the measurement result of glucose concentration to CGMS.
```
